src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -39,13 +36,22 @@ public List<Frequency> getFrequencies() {
         return frequencyRepository.findAll();
     }
 
-    // FIND BY ID
     @Override
-    public Optional<Meal> findMealById(int mealId) {
-        return mealRepository.findById(mealId);
+    public Meal findMealById(int mealId) {
+        Optional<Meal> result = mealRepository.findById(mealId);
+
+        Meal meal = null;
+
+        if (result.isPresent()){
+                meal = result.get();
+        } else {
+            throw new ResponseStatusException(NOT_FOUND, "Invalid meal ID");
```

Comment:
Ah I see why you got rid of the optional . . . Right I would suggest putting the optional back and letting the controller decide what an empty optional should mean. Basically this line is Controller Logic NOT service logic.

So in the controller which calls this method you throw this exception